classes/Cart.php

```diff
@@ -4008,6 +4008,13 @@ public function getRawSummaryDetails(int $id_lang, bool $refresh = false): array
             'invoice' => AddressFormat::getFormattedLayoutData($invoice),
         ];
 
+        /*
+         * First add all applicable cartRules before calculating the totals and the discounts
+         */
+        if (CartRule::isFeatureActive() && $this->id) {
+            $context->cart = $this;
```

Comment:
Did you notice that a lot of functions in Cart.php do
```
$context = Context::getContext()->cloneContext();
$context->cart = $this;
```
Maybe this is in order to avoid side effects when doing
```
$context->cart = $this;
```

Reply:
If you do this you're not cloning anymore, the reference is the same.

Reply:
The cart reference is the same, but not the context one That's the idea, so you can perform "changes" on this new context with fewer issues I guess... The context was already a problem back in the days ^^